src/components/SelectComponents/SuggestionsList/__tests__/SuggestionsList.spec.tsx

```diff
@@ -0,0 +1,92 @@
+import React from 'react';
+import '@testing-library/jest-dom';
+import { render, screen, RenderResult } from '@testing-library/react';
+import { SuggestionsList } from '..';
+import {
+    SUGGESTIONS,
+    SUGGESTION_TO_STRING,
+} from '../../../AutosuggestDeprecated/__mocks__/suggestions';
+
+const noSuggestionsPlaceholder = 'noSuggestionsPlaceholder';
+const getItemProps = (props) => ({ disabled: props.disabled });
+const highlightedIndex = 1;
+const defaultProps = {
+    isLoading: false,
+    suggestions: SUGGESTIONS,
+    suggestionToString: SUGGESTION_TO_STRING,
+    noSuggestionsPlaceholder,
+    getItemProps,
+    highlightedIndex,
+};
+
+let view: RenderResult;
+
+beforeEach(() => {
+    view = render(<SuggestionsList {...defaultProps} />);
+});
+
+const rerenderView = (props) => {
+    view.rerender(<SuggestionsList {...defaultProps} {...props} />);
+};
+
+describe('SuggestionsList', () => {
+    it('should initially render component correctly', () => {
+        expect(view.container).toMatchSnapshot();
+    });
+
+    it('should render empty component correctly', () => {
+        rerenderView({ suggestions: [] });
+
+        expect(screen.getByRole('listitem')).toBeInTheDocument();
+    });
+
+    it('should render optimization list correctly', () => {
+        rerenderView({ useOptimizeRender: true });
+
+        expect(view.container).toMatchSnapshot();
+    });
+
+    it('should render noSuggestions placeholder when empty suggestions list is passed', () => {
+        rerenderView({ suggestions: [] });
+
+        expect(screen.getByRole('listitem').textContent).toEqual(noSuggestionsPlaceholder);
+    });
+
+    it('should render isLoading state', () => {
+        rerenderView({ isLoading: true });
+
+        expect(view.container).toMatchSnapshot();
```

Comment:
there is no expectation for what the test name describes.

Reply:
:+1: